airflow/jobs/scheduler_job.py

```diff
@@ -901,6 +901,33 @@ def __get_concurrency_maps(
             task_map[(dag_id, task_id)] = count
         return dag_map, task_map
 
+    def _remove_tis_with_missing_dag(self, task_instances, session=None):
+        """
+        Fail task instances and the corresponding DagRun if the dag can't be found in
+        the dags folder but exists in SerializedDag table.
+        Return task instances that exists in SerializedDag table as well as dags folder.
+        If the dag can't be found in DagBag, just return the task instance. This is common in
+        unittest where subdir is os.devnull
+        """
+        tis = []
```

Comment:
This is the list of TaskInstances that *will* be queued, right? Could use a better variable name (and a `:returns:` line in the docstring).

I may also turn this into an iterator or even filter function instead to get rid of this local list, so at the call site we can do

```python
task_instances_to_examine = list(
    self._filter_task_instances_to_examine(
        task_instances_to_examine, session=session,
    )  # This is a generator that yields each ti.
)
```

or

```python
task_instances_to_examine = [
    ti for ti in task_instances_to_examine
    if self._mark_task_instance_should_be_examined(ti, session=session)
]
```

Reply:
Thanks @uranusjr, I have changed this implementation to run at regular intervals. Please review